common/data_refinery_common/file_management.py

```diff
@@ -0,0 +1,178 @@
+# used instead.
+def get_temp_dir(batch: Batch, dir_name: str = None) -> str:
+    dir_name = dir_name if dir_name is not None else str(batch.id)
+    return os.path.join(LOCAL_ROOT_DIR,
+                        TEMP_PREFIX,
+                        batch.internal_location,
+                        dir_name)
+
+
+def get_temp_download_path(batch: Batch, dir_name: str = None) -> str:
+    """Returns the path of the downloaded file in the temp directory
+
+    In cases where extraction is necessary, this will not match the
+    name of the batch's extracted file.
+    """
+    return os.path.join(get_temp_dir(batch, dir_name),
+                        _get_downloaded_name(batch))
+
+
+def get_temp_pre_path(batch: Batch, dir_name: str = None) -> str:
+    """Returns the path of the pre-processed file for the batch."""
+    return os.path.join(get_temp_dir(batch, dir_name),
+                        batch.name)
+
+
+def get_temp_post_path(batch: Batch, dir_name: str = None) -> str:
+    """Returns the path of the post-processed file for the batch."""
+    return os.path.join(get_temp_dir(batch, dir_name),
+                        _get_processed_name(batch))
+
+
+def get_processed_dir(batch: Batch) -> str:
+    if USE_S3:
+        return os.path.join(PROCESSED_PREFIX, batch.internal_location)
+    else:
+        return os.path.join(LOCAL_ROOT_DIR, PROCESSED_PREFIX, batch.internal_location)
+
+
+def get_processed_path(batch: Batch) -> str:
+    return os.path.join(get_processed_dir(batch),
+                        _get_processed_name(batch))
+
+
+def upload_raw_file(batch: Batch, dir_name: str = None) -> None:
+    """Moves the batch's raw file out of the temp directory.
+
+    Depending on the value of the USE_S3 environment variable this may
+    just be to the RAW_PREFIX directory or it may be to S3.
+    """
+    temp_path = get_temp_pre_path(batch, dir_name)
+    raw_path = get_raw_path(batch)
+    if USE_S3:
+        bucket = boto3.resource("s3").Bucket(S3_BUCKET_NAME)
+        with open(temp_path, 'rb') as temp_file:
+            bucket.put_object(Key=raw_path, Body=temp_file)
+    else:
+        shutil.copyfile(temp_path, raw_path)
+
+
+def download_raw_file(batch: Batch, dir_name: str = None) -> None:
+    """Moves the batch's raw file to the temp directory.
+
+    Depending on the value of the USE_S3 environment variable this may
+    just be from the RAW_PREFIX directory or it may be from S3.
+    """
+    raw_path = get_raw_path(batch)
+    temp_dir = get_temp_dir(batch, dir_name)
+    os.makedirs(temp_dir, exist_ok=True)
+    temp_path = get_temp_pre_path(batch, dir_name)
+    if USE_S3:
+        bucket = boto3.resource("s3").Bucket(S3_BUCKET_NAME)
+        with open(temp_path, 'wb') as temp_file:
+            bucket.download_fileobj(raw_path, temp_file)
+    else:
+        shutil.copyfile(raw_path, temp_path)
+
+
+def upload_processed_file(batch: Batch, dir_name: str = None) -> None:
```

Comment:
I agree these two functions are similar, but honestly I like that every operation this namespace performs has its own function. I think it'll make it more clear to users of these functions which function they want and what they should pass it. If I tried to merge them then I'd need an additional parameter to differentiate whether to upload a raw or a processed file, but only with those two options. So that parameter could either be a string which I'd have to validate is only `"raw"` or `"processed"` or I could make it a boolean. I there's a bit of code duplication here, which I think I could solve by making them both call a helper function `upload_file` which would just take a from path and a to path. I think that actually makes sense, but I do think I like the readability/usability of: `upload_processed_file(batch)` over `upload_file(batch, raw=False)`.